paramiko_expect.py

```diff
@@ -188,7 +187,7 @@ def expect(
 
             # Add the currently read buffer to the output
             self.current_output += current_buffer_decoded
-            current_buffer_output_decoded = '\n' + self.current_output.splitlines()[-1]
+            current_buffer_output_decoded = '\n' + current_buffer_decoded
```

Comment:
Check the latest output for any matches instead of all the total output - Requires `current_output_line = 0` to be initialized before the while loop
```suggestion
            # Add the currently read buffer to the output
            self.current_output += current_buffer_decoded
            output_lines = self.current_output.splitlines()
            # Check the latest output for any matches
            current_buffer_output_decoded = '\n' + '\n'.join(output_lines[current_output_line:])
            # Update current line in output that has been checked
            current_output_line += len(output_lines[current_output_line:]) - 1
```